vulcanai2/models/basenetwork.py

```diff
@@ -89,14 +96,46 @@ def __init__(self, name, dimensions, config, save_path=None, input_network=None,
             validation_accuracy=[]
         )
 
-        # self._itr = 0 #TODO: ?
+        self.in_dim = self._dimensions
+        if self._input_networks:
+            for i, in_net in enumerate(self._input_networks):
+                if in_net.__class__.__name__ == "ConvNet":
+                    if in_net.conv_flat_dim != self.in_dim:
+                        self.in_dim[i] = self.get_flattened_size(in_net)
+                    else:
+                        pass
 
+                if in_net.__class__.__name__ == "DenseNet":
+                    if in_net.dims[-1] != self.in_dim:
+                        self.in_dim = in_net.dims[-1]
+                    else:
+                        pass
         self._create_network(
             activation=activation,
             pred_activation=pred_activation)
 
     # TODO: where to do typechecking... just let everything fail?
 
+    @abc.abstractmethod
+    def _forward(self):
+        pass
+
+    def forward(self, inputs):
+        """Perform a forward pass through the module/modules."""
+
+        if self._input_networks:
```

Comment:
this check should be done much earlier (e.g.  __init__)

Reply:
mmmm, only if we don't want to allow people to set input networks after the object is initialized. do we? thoughts?

Reply:
yeah I think we should be able to add an input network after it's been initialized, or add more